tests/unit/adapters/application-test.js

```diff
@@ -191,153 +286,239 @@ test('#updateResource', function(assert) {
-  let relationURL = 'http://api.pixelhandler.com/api/v1/posts/1/relationships/author';
-  let jsonBody = '{"data":{"type":"authors","id":"1"}}';
-  let msg = '#fetch called with url and options with data';
-  assert.ok(adapter.fetch.calledWith(relationURL, { method: 'PATCH', body: jsonBody }), msg);
+  promise.then(() => {
+    let jsonBody = JSON.stringify({data: {type: 'authors', id: '1'}});
+    assert.ok(
+      adapter.fetch.calledWith(
+        postMock.data.relationships.author.links.self,
+        {method: 'PATCH', body: jsonBody}
+      ),
+      '#fetch called with url and options with data'
+    );
+    done();
+  });
 });
 
 test('#deleteResource can be called with a string as the id for the resource', function(assert) {
+  assert.expect(2);
+  const done = assert.async();
+
   const adapter = this.subject({type: 'posts', url: '/posts'});
   sandbox.stub(adapter, 'fetch', function () { return RSVP.Promise.resolve(null); });
   let promise = adapter.deleteResource('1');
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  let msg = '#fetch called with url';
-  assert.ok(adapter.fetch.calledWith('/posts/1', { method: 'DELETE' }), msg);
+  promise.then(() => {
+    assert.ok(
+      adapter.fetch.calledWith('/posts/1', {method: 'DELETE'}),
+      '#fetch called with url'
+    );
+    done();
+  });
 });
 
 test('#deleteResource can be called with a resource having a self link, and calls resource#destroy', function(assert) {
+  assert.expect(3);
+  const done = assert.async();
+
   const adapter = this.subject({type: 'posts', url: '/posts'});
   sandbox.stub(adapter, 'fetch', function () { return RSVP.Promise.resolve(null); });
   let resource = this.container.lookup('model:post').create(postMock.data);
   sandbox.stub(resource, 'destroy', function () {});
   let promise = adapter.deleteResource(resource);
+
   assert.ok(typeof promise.then === 'function', 'returns a thenable');
-  assert.ok(resource.destroy.calledOnce, 'resource#destroy method called');
-  let selfURL = 'http://api.pixelhandler.com/api/v1/posts/1';
-  let msg = '#fetch called with url';
-  assert.ok(adapter.fetch.calledWith(selfURL, { method: 'DELETE' }), msg);
+  promise.then(() => {
+    assert.ok(resource.destroy.calledOnce, 'resource#destroy method called');
+    assert.ok(
+      adapter.fetch.calledWith(postMock.data.links.self, {method: 'DELETE'}),
+      '#fetch called with url'
+    );
+    done();
+  });
 });
 
 test('when called with resource argument, #deleteResource calls #cacheRemove', function(assert) {
+  assert.expect(1);
+
   const adapter = this.subject({type: 'posts', url: '/posts'});
   sandbox.stub(adapter, 'fetch', function () { return RSVP.Promise.resolve(null); });
   let resource = this.container.lookup('model:post').create(postMock.data);
   sandbox.stub(adapter, 'cacheRemove', function () {});
-  Ember.run(function() {
+  Ember.run(() => { // Why is this called inside a Ember RunLoop? Does that imply async?
```

Comment:
Since theoretically stuff could break within the runloop I'll use another `assert.expect()` and `async()` here so we know when it does.